diff --git a/src/config.c b/src/config.c
index a94fe60..deba28b 100644
--- a/src/config.c
+++ b/src/config.c
@@ -54,6 +54,8 @@ static struct {
 
 	{ "scroll_down", "e", "Scroll down key.", OPT_KEY },
 	{ "scroll_up", "r", "Scroll up key.", OPT_KEY },
+	{ "scroll_right", "o", "Scroll right key.", OPT_KEY },
+	{ "scroll_left", "i", "Scroll left key.", OPT_KEY },
 
 	{ "cursor_color", "#FF4500", "The color of the pointer in normal mode (rgba hex value).", OPT_STRING },
 
diff --git a/src/normal.c b/src/normal.c
index e23fa74..4502fa9 100644
--- a/src/normal.c
+++ b/src/normal.c
@@ -88,6 +88,8 @@ struct input_event *normal_mode(struct input_event *start_ev, int oneshot)
 		"screen",
 		"scroll_down",
 		"scroll_up",
+		"scroll_right",
+		"scroll_left",
 		"start",
 		"top",
 		"up",
@@ -154,6 +156,22 @@ struct input_event *normal_mode(struct input_event *start_ev, int oneshot)
 				scroll_accelerate(SCROLL_UP);
 			} else
 				scroll_decelerate();
+		} else if (config_input_match(ev, "scroll_right")) {
+			redraw(scr, mx, my, 1);
+
+			if (ev->pressed) {
+				scroll_stop();
+				scroll_accelerate(SCROLL_RIGHT);
+			} else
+				scroll_decelerate();
+		} else if (config_input_match(ev, "scroll_left")) {
+			redraw(scr, mx, my, 1);
+
+			if (ev->pressed) {
+				scroll_stop();
+				scroll_accelerate(SCROLL_LEFT);
+			} else
+				scroll_decelerate();
 		} else if (config_input_match(ev, "accelerator")) {
 			if (ev->pressed)
 				mouse_fast();
diff --git a/src/scroll.c b/src/scroll.c
index 3416c9d..a85137c 100644
--- a/src/scroll.c
+++ b/src/scroll.c
@@ -81,7 +81,13 @@ void scroll_accelerate(int _direction)
 	if (v == 0) {
 		d = 0;
 		traveled = 0;
-		v = v0;
+
+		int scroll_faster = 1;
+		if (direction == SCROLL_RIGHT || direction == SCROLL_LEFT) {
+			scroll_faster = 4;
+		}
+
+		v = v0 * scroll_faster;
 	}
 }
 

