diff --git a/src/platform/linux/X/input.c b/src/platform/linux/X/input.c
index ba3aca5..9f75fac 100644
--- a/src/platform/linux/X/input.c
+++ b/src/platform/linux/X/input.c
@@ -366,7 +366,7 @@ struct input_event *x_input_wait(struct input_event *events, size_t sz)
 {
 	size_t i;
 	static struct input_event ev;
-	struct input_evnet *ret = NULL;
+	struct input_event *ret = NULL;
 
 	for (i = 0; i < sz; i++) {
 		struct input_event *ev = &events[i];
