diff --git a/config.h b/config.h
index 575f11c..e0639b8 100644
--- a/config.h
+++ b/config.h
@@ -11,11 +11,13 @@ static const char col_gray1[]       = "#222222";
 static const char col_gray2[]       = "#444444";
 static const char col_gray3[]       = "#bbbbbb";
 static const char col_gray4[]       = "#eeeeee";
-static const char col_cyan[]        = "#005577";
-static const char *colors[][3]      = {
-	/*               fg         bg         border   */
-	[SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
-	[SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
+static const char col_cyan[]        = "#ef0495";
+static const char col_blue1[]        = "#626f95";
+static const char col_blue4[]        = "#4c73e4";
+static const char *colors[][4]      = {
+	/*               fg         bg         border     float */
+	[SchemeNorm] = { col_gray3, col_gray1, col_gray2, col_blue1 },
+	[SchemeSel]  = { col_gray4, col_cyan,  col_cyan, col_blue4 },
 };
 
 /* tagging */
