diff --git a/config.h b/config.h
index 56ee471..ca8f655 100644
--- a/config.h
+++ b/config.h
@@ -1,9 +1,9 @@
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-static const unsigned int borderpx  = 1;        /* border pixel of windows */
+static const unsigned int borderpx  = 5;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
-static const int showbar            = 1;        /* 0 means no bar */
+static const int showbar            = 0;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
 static const char *fonts[]          = { "monospace:size=10" };
 static const char dmenufont[]       = "monospace:size=10";
@@ -28,9 +28,27 @@ static const Rule rules[] = {
 	 *	WM_CLASS(STRING) = instance, class
 	 *	WM_NAME(STRING) = title
 	 */
-	/* class      instance    title       tags mask     isfloating   monitor */
-	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
+	/* class                  instance                title                      tags mask     isfloating   monitor    float x,y,w,h    floatborderpx       unmanaged */
+	{ "Plasma",               NULL,                   NULL,                      0,            1,           -1,        -1,-1,-1,-1,     -1,                  1 }, // make plasma windows unmanaged
+	{ "plasma-desktop",       NULL,                   NULL,                      0,            1,           -1,        -1,-1,-1,-1,     -1,                  1 }, // make plasma windows unmanaged
+	{ "plasmashell",          NULL,                   NULL,                      0,            1,           -1,        -1,-1,-1,-1,     -1,                  1 }, // make plasma windows unmanaged
+	{ "plasmashell",          "plasmashell",          "Desktop @ QRect",         1 << 8,       0,           -1,        -1,-1,800,800,   -1                     }, // send plasma desktop to tag 9
+	{ "kitty-dmenu",          "kitty-dmenu",          NULL,                      0,            1,           -1,        50,150,1800,800,  -1                     }, // float and size kitty-dmenu window
+	{ "jdownloader",          "jdownloader",          "JDownloader 2",           1 << 8,       0,           -1,        -1,-1,-1,-1,     -1                     }, // send jdownloader to tag 9
+	{ NULL,                   NULL,                   "is sharing your screen.", 1 << 8,       1,           -1,        -1,-1,-1,-1,     -1                     }, // send google meet popup to tag 9
+	{ "kcalc",                "kcalc",                NULL,                      0,            1,           -1,        -1,-1,-1,-1,     -1                     }, // make calculator floating by default
+	{ "mouseless",            "Mouseless",            "Mouseless",               0,            1,           -1,        -1,-1,-1,-1,     -1                     }, // make mouseless-click config floating by default
+	{ "kronometer",           "kronometer",           NULL,                      0,            1,           -1,        -1,-1,1800,800,  -1                     }, // make nnn picker floating by default
+	{ "Google-chrome",        "google-chrome",        NULL,                      1 << 4,       0,           -1,        -1,-1,-1,-1,     -1                     }, // use tag 5 for chrome
+	{ "Microsoft-edge",       "microsoft-edge",       NULL,                      1 << 4,       0,           -1,        -1,-1,-1,-1,     -1                     }, // use tag 5 for edge
+	{ "Chromium",             "chromium",             NULL,                      1 << 4,       0,           -1,        -1,-1,-1,-1,     -1                     }, // use tag 5 for chromium
+	{ "Bitwarden",            "bitwarden",            "Bitwarden",               1 << 8,       0,           -1,        -1,-1,-1,-1,     -1                     }, // use tag 9 for bitwarden
+	{ NULL,                   NULL,                   "NoiseTorch",              1 << 8,       0,           -1,        -1,-1,-1,-1,     -1                     }, // use tag 3 for noisetorch
+	{ "MyPaint",              "mypaint",              NULL,                      1 << 5,       0,           -1,        -1,-1,-1,-1,     -1                     }, // use tag 6 for mypaint
+	{ "SimpleScreenRecorder", "simplescreenrecorder", "SimpleScreenRecorder",    0,            1,           -1,        -1,-1,-1,-1,     -1                     },
+	{ NULL,                   NULL,                   "KRunner",                 0,            1,           -1,        -1,-1,-1,-1,     -1                     },
+	{ NULL,                   NULL,                   "Event Tester",            0,            1,           -1,        -1,-1,-1,-1,     -1                     },
+
 };
 
 /* layout(s) */
@@ -48,7 +66,7 @@ static const Layout layouts[] = {
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
@@ -61,12 +79,11 @@ static const Layout layouts[] = {
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
-static const char *termcmd[]  = { "st", NULL };
+static const char *termcmd[]  = { "kitty", NULL };
 
 static const Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
-	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
@@ -118,4 +135,3 @@ static const Button buttons[] = {
 	{ ClkTagBar,            MODKEY,         Button1,        tag,            {0} },
 	{ ClkTagBar,            MODKEY,         Button3,        toggletag,      {0} },
 };
-
