diff --git a/packages/react-devtools-shared/src/utils.js b/packages/react-devtools-shared/src/utils.js
index bc8e768..02025d3 100644
--- a/packages/react-devtools-shared/src/utils.js
+++ b/packages/react-devtools-shared/src/utils.js
@@ -507,7 +507,7 @@ export function getDefaultPreset(): 'custom' | 'vscode' {
 export function getDefaultOpenInEditorURL(): string {
   return typeof process.env.EDITOR_URL === 'string'
     ? process.env.EDITOR_URL
-    : vscodeFilepath;
+    : 'terminal://{path}:{line}';
 }
 
 export function getOpenInEditorURL(): string {
