diff --git a/PKGBUILD b/PKGBUILD
index a60525d..f7a7add 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,7 +13,7 @@ makedepends=(python-build python-installer python-setuptools python-wheel)
 optdepends=('python-freesimplegui: for rtcqs_gui GUI program')
 groups=(pro-audio)
 source=("$pkgname-$pkgver.tar.gz::https://codeberg.org/rtcqs/$pkgname/archive/v$pkgver.tar.gz")
-sha256sums=('4b4431f2377f5f6b151856c154daf1e4d024a4d801da7aca722972e01b2c926b')
+sha256sums=('22bd9798c24e99351c259069e136b8683c211abc551775595cbacc3c8a19eae7')
 
 build() {
   cd $pkgname
