diff --git a/lua/trailblazer/trails/common.lua b/lua/trailblazer/trails/common.lua
index 5c31c69..8d158f7 100644
--- a/lua/trailblazer/trails/common.lua
+++ b/lua/trailblazer/trails/common.lua
@@ -219,24 +219,12 @@ end
 ---@param ext_mark table
 ---@return boolean
 function Common.focus_win_and_buf(trail_mark, ext_mark)
-  local ok
-  local win_ids = api.nvim_list_wins()
-
-  if trail_mark.win == nil or not vim.tbl_contains(win_ids, trail_mark.win) then
-    trail_mark.win = win_ids[1] or api.nvim_get_current_win()
-  end
-
-  ok, _ = pcall(api.nvim_set_current_win, trail_mark.win)
-  if not ok then
-    api.nvim_set_current_win(0)
-  end
-
   if trail_mark.buf == nil or not api.nvim_buf_is_loaded(trail_mark.buf) or
       not api.nvim_buf_is_valid(trail_mark.buf) then
     return false
   end
 
-  ok, _ = pcall(api.nvim_set_current_buf, trail_mark.buf)
+  local ok, _ = pcall(api.nvim_set_current_buf, trail_mark.buf)
   if ok then
     ok, _ = pcall(api.nvim_win_set_cursor, 0, { ext_mark[1][2] + 1, ext_mark[1][3] })
     if ok then
