diff --git a/src/nvim/drawline.c b/src/nvim/drawline.c
index 838f603060..b1b25d9937 100644
--- a/src/nvim/drawline.c
+++ b/src/nvim/drawline.c
@@ -519,10 +519,8 @@ void fill_foldcolumn(win_T *wp, foldinfo_T foldinfo, linenr_T lnum, int attr, in
       symbol = wp->w_p_fcs_chars.foldsep;
     } else if (wp->w_p_fcs_chars.foldinner != NUL) {
       symbol = wp->w_p_fcs_chars.foldinner;
-    } else if (first_level + i <= 9) {
-      symbol = schar_from_ascii('0' + first_level + i);
     } else {
-      symbol = schar_from_ascii('>');
+      symbol = schar_from_ascii(' ');
     }
 
     int vcol = i >= level ? -1 : (i == closedcol - 1 && closed) ? -2 : -3;
