diff --git a/lua/overseer/strategy/jobstart.lua b/lua/overseer/strategy/jobstart.lua
index 3007710..72e2f43 100644
--- a/lua/overseer/strategy/jobstart.lua
+++ b/lua/overseer/strategy/jobstart.lua
@@ -321,6 +321,7 @@ function JobstartStrategy:start(task)
   })
 
   self.job_id = overseer.builtin.jobstart(task.cmd, opts)
+  task.job_id = self.job_id
 
   if self.job_id == 0 then
     log.error("Invalid arguments for task '%s'", task.name)
