diff --git a/lua/overseer/layout.lua b/lua/overseer/layout.lua
index 8ac50a4..db4a621 100644
--- a/lua/overseer/layout.lua
+++ b/lua/overseer/layout.lua
@@ -19,7 +19,7 @@ M.get_editor_width = function()
 end
 
 M.get_editor_height = function()
-  local editor_height = vim.o.lines - vim.o.cmdheight
+  local editor_height = vim.o.lines
   -- Subtract 1 if tabline is visible
   if vim.o.showtabline == 2 or (vim.o.showtabline == 1 and #vim.api.nvim_list_tabpages() > 1) then
     editor_height = editor_height - 1
