diff --git a/src/background/index.ts b/src/background/index.ts
index 38d3e9b..442bbba 100644
--- a/src/background/index.ts
+++ b/src/background/index.ts
@@ -26,15 +26,6 @@ export async function init() {
     'activate',
   ];
 
-  chromeAPI.runtime.onInstalled.addListener(async installDetails => {
-    // Check whether this is the inital installation or an update of an existing
-    // installation.
-    if (installDetails.reason === 'install') {
-      const extensionInfo = await chromeAPI.management.getSelf();
-      extension.firstRun(extensionInfo);
-    }
-  });
-
   // Respond to messages sent by the JavaScript from https://hyp.is.
   // This is how it knows whether the user has this Chrome extension installed.
   chromeAPI.runtime.onMessageExternal.addListener(
