diff --git a/src/background/extension.ts b/src/background/extension.ts
index 95de324..d3201e8 100644
--- a/src/background/extension.ts
+++ b/src/background/extension.ts
@@ -161,8 +161,8 @@ export class Extension {
           tab.status === 'complete' || typeof tab.status !== 'string';
 
         state.setState(tab.id, {
-          state: isActive ? 'active' : 'inactive',
-          extensionSidebarInstalled: isActive,
+          state: tab.url?.includes("youtube.com")  ? 'inactive' : 'active',
+          extensionSidebarInstalled: true,
           ready,
         });
       }
@@ -181,6 +181,7 @@ export class Extension {
         return;
       }
 
+      current.state = tab.url?.includes("youtube.com") ? 'inactive' : 'active';
       browserAction.update(tabId, current);
 
       addOrRemoveClientFromTab(tab);
@@ -266,9 +267,7 @@ export class Extension {
         currentlyLoadingUrl.delete(tabId);
         const tabState = state.getState(tabId);
         let newActiveState = tabState.state;
-        if (tabState.directLinkQuery) {
-          newActiveState = 'active';
-        }
+        newActiveState = tab.url?.includes('youtube.com') ? 'inactive' : 'active';
         state.setState(tabId, {
           ready: true,
           state: newActiveState,
